### PR TITLE
chore(plugins): Convert plugin resource views to use $vars instead of input

### LINKED ISSUE #8754: Convert resource views to use `$vars` (plugins)
Companion to #8569

### PR DESCRIPTION
**Pull request by mrclay**
Also moved uservalidationbyemail confirm page to resource view.

Fixes #8754

### CODE REVIEW

File: mod/pages/views/default/resources/pages/edit.php [hunk 1]
```diff
@@ -30,15 +29,7 @@
 
 if ($page->canEdit()) {
 
-	if ($revision) {
-		$revision = elgg_get_annotation_from_id($revision);
-		if (!$revision || !($revision->entity_guid == $page_guid)) {
-			register_error(elgg_echo('pages:revision:not_found'));
-			forward(REFERER);
-		}
-	}
```
Comment (ewinslow) [inline mod/pages/views/default/resources/pages/edit.php]: This wasn't needed at all?

File: mod/uservalidationbyemail/start.php [hunk 2]
```diff
@@ -197,7 +197,7 @@ function uservalidationbyemail_check_auth_attempt($credentials) {
 function uservalidationbyemail_page_handler($page) {
 	switch ($page[0]) {
 		case 'confirm':
-			require __DIR__ . "/pages/confirm.php";
+			echo elgg_view_resource("uservalidationbyemail/confirm");
```
Comment (ewinslow) [inline mod/uservalidationbyemail/start.php]: Huh... we missed one?

**Comment by ewinslow** [comment]
LGTM

**Comment by ewinslow** [comment]
not tested.

File: mod/pages/views/default/resources/pages/edit.php (hunk 1 above)
Comment (mrclay) [inline mod/pages/views/default/resources/pages/edit.php]: We no longer allow editing specific revisions. At least there's UI that leads to it.

File: mod/uservalidationbyemail/start.php (hunk 2 above)
Comment (mrclay) [inline mod/uservalidationbyemail/start.php]: Yeah.

File: mod/pages/views/default/resources/pages/edit.php (hunk 1 above)
Comment (mrclay) [inline mod/pages/views/default/resources/pages/edit.php]: No UI in last 1.8 version either.

**Pull request merged**